### ISSUE TITLE
Avoid ENV inside Dockerfile, mainly with sensible data like passwords

### ISSUE THREAD

**Issue by mfandrade** [hadolint/hadolint#462]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

```
$ cat <<EOF > Dockerfile
FROM mariadb:10
ENV MYSQL_ROOT_PASSWORD=my-secret-pw
EOF

$ hadolint Dockerfile
Dockerfile:2 DL???? Explicit env value defined inside Dockerfile
```
OBS: Sensible env vars must be passed to Docker in runtime.
```
$ docker container run -e MYSQL_ROOT_PASSWORD=my-secret-pw ...
```

### Actual behavior

There is no such rule.

### Steps to reproduce the behavior

N/A

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.17.4-0-g43bca62
```

**Dockerfile** (if relevant)

N/A

**Additional environment details (OS, stack version, etc.)**

N/A
